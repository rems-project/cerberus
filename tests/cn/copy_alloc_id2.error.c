void f (int *p)
/*@
requires
    has_alloc_id(p);
ensures
    true;
@*/
{
  unsigned long long p_int = (unsigned long long) p;
  int* q = __cerbvar_copy_alloc_id(p_int + 0ULL, p);
  /*@ assert (ptr_eq(p, q)); @*/
}

int main(void)
{
  int p[1] = {1};
  f(p);
}
