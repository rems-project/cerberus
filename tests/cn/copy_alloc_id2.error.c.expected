return code: 1
[1/2]: f -- fail
[2/2]: main -- pass
tests/cn/copy_alloc_id2.error.c:10:12: error: Pointer `p` needs to be live for copy_alloc_id
  int* q = __cerbvar_copy_alloc_id(p_int + 0ULL, p);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 
Need an Alloc or Owned in context with same allocation id
State file: file:///tmp/state__copy_alloc_id2.error.c__f.html
